bears/coffee_script/CoffeeLintBear.py

```diff
@@ -22,11 +23,259 @@ class CoffeeLintBear:
+        :param: no_interpolation_in_single_quotes:
+            Prohibits string interpolation in a single quoted string.
+
+            Example: If ``no_interpolation_in_single_quotes = True`` then
+            ```
+            f = '#{bar}'
+            ``` is prohibited whereas
+            ```
+            f = "#{bar}"
+            ``` is accepted.
+        :param: no_stand_alone_at:
+            Checks if stand alone at ``@`` are in use.
+        :param: no_throwing_strings:
+           Prohibits throwing string literals or interpolation.
+        :param: no_trailing_semicolons:
+           Prohibits trailing semicolons since they are not useful.
+
+           Example: If ``no_trailing_semicolon = True``
+           ```
+           x = '1234'; console.log(x)
+           ``` here the semicolon is meaningfull.
+           ```
+           alert('end of line');
+           ``` the semicolon is useless.
+        :param: no_trailing_whitespaces:
+            Checks if they are trailing whitespace in the code.
+        :param: apply_no_trailing_whitespaces_in_comments:
+            Determines if ``no_trailing_whitespace`` rule should be applied to
+            comments.
+        :param: apply_no_trailing_whitespaces_in_empty_line:
+            Determines if ``no_trailing_whitespace`` rule should be applied to
+            empty lines.
+        :param: no_unecessary_double_quotes:
+            Prohibits double quotes unless string interpolation is used or the
+            string contains single quotes.
+        :param: use_english_operator:
+            Determines if ``and``, ``or``, ``is`` and ``isnt`` should be use
+            instead of ``&&``, ``||``, ``==`` and ``!=``.
+        :param: space_around_operators:
+            Enforces that operators have spaces around them.
+        :param: space_after_comma:
+            Checks if there is a space after commas.
+        :param: cyclomatic_complexity:
+            Determines if the complexity of the file should be examine.
+        :param: cyclomatic_complexity_value:
+            Cyclomatic complexity of the file.
+        :param: duplicate_keys:
+            Prevents defining duplicate keys in object literals and classes.
+        :param: enforce_parentheses_on_empty_constructor:
+            Requires constructors with no parameters to include parentheses.
+        :param: line_endings:
+            Ensures that the file uses only ``windows`` or ``unix`` line
+            endings.
+        :param: line_endings_option:
+            The option to ``line_endings``, its value is either ``unix`` or
+            ``windows``.
+        :param: no_this:
+            prohibits ``this``. Use ``@`` instead.
+        :param: no_plusplus:
+            Forbids the increment and decrement arithmetic operators.
+        :param: no_empty_parameter_list:
+            Prohibits empty parameter lists in function definitions.
+        :param: no_empty_functions:
+            Disallows declaring empty functions.
```

Comment:
thanks, :) this is a good one :)

2016-06-27 16:29 GMT+01:00 Zatreanu Adrian-Gabriel <notifications@github.com
>:

> In bears/coffee_script/CoffeeLintBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/555#discussion_r68597466>
> :
>
> > +        :param: enforce_parentheses_on_empty_constructor:
> > +            Requires constructors with no parameters to include parentheses.
> > +        :param: line_endings:
> > +            Ensures that the file uses only ``windows`` or ``unix`` line
> > +            endings.
> > +        :param: line_endings_option:
> > +            The option to ``line_endings``, its value is either ``unix`` or
> > +            ``windows``.
> > +        :param: no_this:
> > +            prohibits ``this``. Use ``@`` instead.
> > +        :param: no_plusplus:
> > +            Forbids the increment and decrement arithmetic operators.
> > +        :param: no_empty_parameter_list:
> > +            Prohibits empty parameter lists in function definitions.
> > +        :param: no_empty_functions:
> > +            Disallows declaring empty functions.
>
> use Prohibits to be consistent
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/555/files/b2928984ba5005450f03a258f5293f325f2c5d17#r68597466>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xom0nQL3-Uzebux7mg7rzaIditfhIks5qP-xUgaJpZM4I9f6t>
> .
>